### ISSUE TITLE
Suggestion - Rentable/Purchasable Warehouses for storage

### ISSUE THREAD

**Issue by Sozibro** [InZidiuZ/op-framework-issue-tracker#3271]
### Assurance

I have read the `README.md`

### Summary

In the midst of starting Tony's Supply Store, I find myself in dire need of a storage warehouse.  I currently move high quantities of mechanical materials (steel, aluminum, scrap, glass, and rubber) and it gets very messy and sluggish when I have to juggle rumpos.  These warehouses could be locked exclusively to those materials if that makes it more appealing.  It would require no new game mechanics and there are tons of locations already in Los Santos that could function as the building.  As far as storage, enough to hold 1000 steel, aluminum, and scrap, as well as 400 glass and rubber would be ideal.  That comes out to 17,800su and would be the most I would ever want to have on hand at one time.  The cost could also be on par with other businesses (I believe currently $100,000).  If adding game mechanics was wanted, I have a slew of ideas I can expand on in the comments.

**↓ Further breakdown below  ↓**

Option 1 - Rental Units

    -Static location for everyone to access
    -Tiered system with weekly costs 
        -Tier 1=2,500su=$1,000/week
        -Tier 2=5,000su=$2,250/week
        -Tier 3=10,000su=$5,000/week
    -locked to nonperishable
    -1/person/tier
    -Items confiscated if unpaid for 2 consecutive weeks

Option 2 - Purchasable warehouse

    -Costs $100,000 (current rate to buy a business I believe)
    -Owner must find unique locations (probably a limited number) 
    -20,000su
    -locked to nonperishables
    -no more then one per person
    -same activity time requirements that are in works for business owners or else risk confiscation

Option 3 - Freeze locker

    -Must own restaurant
    -Limited to perishables
    -10,000su
    -Either unique location or attachment to current establishment
    -additional cost tbd
    -extend shelf life

Potenial down the road - High value transport jobs either scripted or RP.

![Business card](https://github.com/InZidiuZ/op-framework-issue-tracker/assets/163086049/7b0cf005-713f-4661-872f-0d6063324e76)

### Reason

For most people houses are enough, but I can't be the only business person in the city that would be interested in purchasing one of these warehouses.  There could be other warehouse types like cold storage for all our new restaurant owners that could hold only food products and maybe extend shelf life.  There are a lot of possibilities for this in multiple industries but I want to keep the scope of this post narrow.

**Comment by scufflord** [comment]
Criminal warehouses could provide some fun additional RP for both sides as well, but I believe there's already a github for a similar idea like #2272

**Comment by Sozibro** [comment]
> Criminal warehouses could provide some fun additional RP for both sides as well, but I believe there's already a github for a similar idea like #2272

Yes! Lots of opportunity!  That one is definitely more involved though and serves a different purpose so hopefully this post is still appropriate.

**Comment by coalaura** [comment]
I really like this idea! Could maybe even have like rentable storage units n stuff, similar to motels

**Comment by Sozibro** [comment]
> I really like this idea! Could maybe even have like rentable storage units n stuff, similar to motels

Yes! I know trailers are a little buggy but rentable storage units that could be loaded and moved would be so cool.

**Comment by TacticalTyl3r** [comment]
Could also make them spots for the Trucking job or Go Postal to "Deliver" stuff there to get more foot traffic.

**Comment by Sozibro** [comment]
> Could also make them spots for the Trucking job or Go Postal to "Deliver" stuff there to get more foot traffic.

I've been thinking along the same thread how, maybe with some city funding, it could create a whole job like in vanilla where there is high value cargo that needs to be moved.   This would create cool opportunity for criminals to get in on the action by hijacking, as well as create a need for hired security on behalf of the owner of the shipment.  Right now I couldn't fund much because my supply is mainly materials but that's where some government funding, maybe gold bars and such, could help create incentive.

**Comment by JennyLynn001** [comment]
I agree this would be neat to have. For years being in legacy it's something myself and others have been wanting.

**Comment by Sozibro** [comment]
I wanted to clarify when I say "it would require no new game mechanics" I was thinking it could be as simple as a static public locker at least to start.

**Comment by coalaura** [comment]
no, i dont wanna add a scuffy half working thing

**Comment by Sozibro** [comment]
Me either?

**Comment by Sozibro** [comment]
Sorry, I was trying to say the opposite like I know it as whole would take a lot and I didn't want to imply it would be easy when I said it would take no new mechanics.  I was wanted offer something I conceived as simple and established instead of from the ground up as a place to start.

**Comment by ayushpatel1007** [comment]
I am not sure but wouldn’t it effect people buying house? I mean if we are providing 10,000 su storage for just $5k per week then I think housing will be last thing they will think to buy. Yea obvious warehouse can be great addition but it should not have extreme large storage’s . Stilll I am not sure as I don’t know final rent amount and storage (SU) for warhouse

**Comment by Sozibro** [comment]
> I am not sure but wouldn’t it effect people buying house? I mean if we are providing 10,000 su storage for just $5k per week then I think housing will be last thing they will think to buy. Yea obvious warehouse can be great addition but it should not have extreme large storage’s . Stilll I am not sure as I don’t know final rent amount and storage (SU) for warhouse

I thought about this and decided it probably would affect it but only slightly.  The amounts of su I am suggesting are the equivalent of just two houses at the most so people will still need houses if they are holding more then 20,000su and anyone holding less would likely still just use one or two houses instead for the sake of having a house/garage/storage at two locations across the map.

**Comment by coalaura** [comment]
So, instead of making storage units i've made something similar which fills the same purpose. You can now rent containers down at the docks for $8,000 a week with 20k storage each. You can give/remove others access and pay rent using /containers which has a similar UI to savings accounts

**Comment by selliee** [comment]
> I am not sure but wouldn’t it effect people buying house? I mean if we are providing 10,000 su storage for just $5k per week then I think housing will be last thing they will think to buy. Yea obvious warehouse can be great addition but it should not have extreme large storage’s . Stilll I am not sure as I don’t know final rent amount and storage (SU) for warhouse

Yes, myself being a realtor will agree with this. The real estate "market" is already dying. 99% of the real estate companies go bankrupt at least 1x a week due to people not sticking around the server for long- causing realtors to be out 2 weeks of rent which could range anywhere from 1k-20k PER WEEK, before we are able to evict them to resell the home, if there is someone who would buy it to begin with. 

As much as I think this idea is a really cool one- I think it should be reserved for WL gangs, and/or WL businesses for giving employees storage such as mechanic shops [however this i would agree would take away from people renting homes solely for mat storage]. I don't see a reason for a single individual to rent a storage unit for 8k a week and get 20k storage when they could interact and support real estate companies by renting houses for the equal storage.

Nobody is going to want to rent a house anymore- and this takes away from real estate RP as well. 

Id honestly support this idea more if there was not 'weekly maintenance fees' on the blocks so there's no loss for realtors when they are not filling unwanted houses due to everyone flocking for a cheaper, more storage rich option to hide their guns in.

**Comment by spitslave** [comment]
> > I am not sure but wouldn’t it effect people buying house? I mean if we are providing 10,000 su storage for just $5k per week then I think housing will be last thing they will think to buy. Yea obvious warehouse can be great addition but it should not have extreme large storage’s . Stilll I am not sure as I don’t know final rent amount and storage (SU) for warhouse
> 
> Yes, myself being a realtor will agree with this. The real estate "market" is already dying. 99% of the real estate companies go bankrupt at least 1x a week due to people not sticking around the server for long- causing realtors to be out 2 weeks of rent which could range anywhere from 1k-20k PER WEEK, before we are able to evict them to resell the home, if there is someone who would buy it to begin with.
> 
> As much as I think this idea is a really cool one- I think it should be reserved for WL gangs, and/or WL businesses for giving employees storage such as mechanic shops [however this i would agree would take away from people renting homes solely for mat storage]. I don't see a reason for a single individual to rent a storage unit for 8k a week and get 20k storage when they could interact and support real estate companies by renting houses for the equal storage.
> 
> Nobody is going to want to rent a house anymore- and this takes away from real estate RP as well.
> 
> Id honestly support this idea more if there was not 'weekly maintenance fees' on the blocks so there's no loss for realtors when they are not filling unwanted houses due to everyone flocking for a cheaper, more storage rich option to hide their guns in.

Completely agree with this, there’s already such lack of RP in the city now people are going to RP less with realtors to purchase rentable whatever’s where they don’t need to talk to anyone. If this becomes a thing I think the realtors should at least be the ones to sell rentable storage units

**Comment by xplrerin** [comment]
> So, instead of making storage units i've made something similar which fills the same purpose. You can now rent containers down at the docks for $8,000 a week with 20k storage each. You can give/remove others access and pay rent using /containers which has a similar UI to savings accounts

I hella supported the idea of warehouses if realtors could be the ones to sell but isn’t this new addition just completely ruining real estate? like said above, people are going to be way more inclined to pay $8k for 20k storage over a house in which only goes 10k storage max and sometimes can cost them $12-15k. Why are they so cheap?

**Comment by coalaura** [comment]
> Why are they so cheap?
Cause its storage not a house, its just an inventory you have no customization, you cant enter them. Houses are not there to be JUST for storage

**Comment by coalaura** [comment]
would be pointless to add storage lockers and then make them cost more than houses

**Comment by ayushpatel1007** [comment]
I also think that this will have negative impact on real estate and people will stop buying house as currently if we see list of owners of house , there are many people buying house for storage also and if this new container storage will come into extisence than it will be create big damage to real estate comapnies. As already there are 5 companies in server with each of them holding 10-15 blocks annd there is no shotage for houses in server.

**Comment by xplrerin** [comment]
> > Why are they so cheap?
> 
> Cause its storage not a house, its just an inventory you have no customization, you cant enter them. Houses are not there to be JUST for storage

Yea Houses have more perks yes but a lot of people will now prefer to own a cheap house, or cheap gang block and the storage unit be their main stash. This doesn’t help the big blocks we continuously have issues with.

**Comment by Alyshiva** [comment]
to be the devils advocate, i have no intention of buying a house, i have no desire to buy one, but i would buy a storage locker for my materials. i have 15+ vans i use for storage now and i know a lot of people like me. the housing market isn't struggling because of storage, its struggling because people don't want to pay 15k every week for something they don't even use.

**Comment by CloudCait** [comment]
Imo this would be better if houses were individually owned by realtors rather than blocks, cos typically the shitty mansions (ones without driveways or gardens) only get purchased for the storage. It’s going to be increasingly difficult to sell the properties that people would typically only buy for storage due to not having a desirable exterior.

On the flip side, I’m intrigued to see what rp this can bring from a cop/fib perspective. Can these containers be robbed, raided etc?

**Comment by coalaura** [comment]
robbed no, raided yes

**Comment by vol5** [comment]
**Realtor Apocalypse - FUNNY EXAGGERATION** @coalaura

I can now get more storage for a fraction of the price and never have to use a realtor again. Yes, it's a cool idea to have storage units, but it's beyond detrimental to the overall housing market in its current form.

The percentage of houses currently owned purely for storage use is grossly underestimated here. This will completely invalidate a solid portion of the housing market. My guess is a 20%-30% drop. Why would I pay an average price of 12k/week for another house when I can get 2 times the storage for 33% less weekly? The value of convenience is king. 20k storage all in one location is massive enough to where anyone who has an extra storage house will drop it within a week of knowing this is a thing.

> Houses are not there just for storage, correct. But completely devaluing one of the main selling points for a house is unreasonable without an equivalent addition to add the lost value.

Five total realtors have voiced their concerns, four from international, one other. If this many people truly don't think its a good idea as is. It should be improved before implementation. 

- typically the shitty mansions (ones without driveways or gardens) only get purchased for the storage.
- Houses have more perks, yes, but a lot of people will now prefer to own a cheap house or cheap gang block and the storage unit be their main stash. This doesn’t help the big blocks we continuously have issues with.
- There are many people buying houses for storage also, and if this new container storage will come into existence, it will create big damage to real estate companies. As already, there are five companies in the server, each holding 10-15 blocks, and there is no shortage of houses in the server.
- People are going to be way more inclined to pay $8k for 20k storage over a house which only goes up to 10k storage max and sometimes can cost them $12-15k. Why are they so cheap?
- Such a lack of RP in the city now; people are going to RP less with realtors to purchase rentable whatever's where they don’t need to talk to anyone.
- I don't see a reason for a single individual to rent a storage unit for 8k a week and get 20k storage when they could interact and support real estate companies by renting houses for the equal storage.

**Recent devaluing of houses/realty:**

- Get more cost-effective storage anytime 24/7
- Grow weed outside

**Recent additions of value for houses/realty:**

- Marginal increase in storage
- Addition of more furniture and higher quantity

The bottom line for me is that it will require zero interaction with other players and take away RP from realtors and buyers alike. I don't find the storage containers in their current form a valuable addition alongside a critical system that is already not doing the best.

**Potential Solutions:**

- Have them be sold by realtors and be more expensive at raw cost.
- Don’t add it.

**Comment by coalaura** [comment]
imma be fr, we used to store all of our items in cars and the realtor market was completely fine

**Comment by spitslave** [comment]
> imma be fr, we used to store all of our items in cars and the realtor market was completely fine

I’m going to be fr and honest not a lot of people maybe the very minimum only store materials into their car but things like guns drugs and that sort of stuff people store in houses because they get older in cars way more. These rentable storages should be sold by realtors that way you still have RP and it doesn’t completely ruin the realtors RP. I don’t know why people should just be able to rent storage without talking to anyone anyway

**Comment by Dibbey** [comment]
As someone who has visited many houses and mansions in the past where there's literally nothing in there except for storage boxes and being a casual person in the city, I can see realtors not really being excited about this.

Why would I pay $9k-$14k for a 10k storage house if I can get double the storage for just $8k. 
So instead of renting at least 2 homes getting a container sounds way more convenient, costs less, all in one spot and I dont have to drive to another second location to access the other 10k su I want or need.

> Houses are not there to be JUST for storage

But the thing is... they are being used for that. Not everyone is actually furnishing their houses. I'm sure if I asked some WL Group Members a lot of them would show me multiple places just used for storage or parking cars. And this _could_, doesn't mean it will, but _could_ result in people dropping those and just getting cheaper containers leaving blocks to possibly be abandoned.
I like the above suggestions of having these sold and maintained by realtors and at least closer to the price-range of current house-to-storage prices which seem to be:

- $9-14k = 9-10k su
- $6-7k = 6k su
- $5.5-6.5k = 5k su
- $6k = 4-4.5k su
- $3-4k = 3500 su
- $2.5-3k = 2000-2500 su

(Information taken from currently listed houses from Close2Home, OnlyHomes and Moonlight Estates)

**Comment by Sozibro** [comment]
If houses were getting used just for storage then this was a long time coming imo.  Feels like all these realtors are really discounting all the people with 10+ rumpos and no houses that would gladly pay an 8k premium to not have to manage all the rumpos.

**Comment by Dibbey** [comment]
I don't think any of us are discounting them. As far as I can tell people generally like this idea, just not the current state of it.

> "As much as I think this idea is a really cool one- (...)"

> "I hella supported the idea of warehouses if realtors could be the ones to sell (...)"

> "On the flip side, I’m intrigued to see what rp this can bring from a cop/fib perspective (...)"

And I'm going to be honest here, I can't lie, I _also_ would drop a house for the sake of paying less and getting more storage. Because in the end that is what some people will do for a cheaper and more convenient option. But if I view it from the POV of a realtor, I don't think I would be excited about that.

**Comment by Sozibro** [comment]
> I don't think any of us are discounting them. As far as I can tell people generally like this idea, just not the current state of it.
> 
> > "As much as I think this idea is a really cool one- (...)"
> 
> > "I hella supported the idea of warehouses if realtors could be the ones to sell (...)"
> 
> > "On the flip side, I’m intrigued to see what rp this can bring from a cop/fib perspective (...)"
> 
> And I'm going to be honest here, I can't lie, I _also_ would drop a house for the sake of paying less and getting more storage. Because in the end that is what some people will do for a cheaper and more convenient option. But if I view it from the POV of a realtor, I don't think I would be excited about that.

I didn't say anything about not liking the idea?

**Comment by selliee** [comment]
![image](https://github.com/InZidiuZ/op-framework-issue-tracker/assets/90532434/1016296a-44cf-4f2f-abd4-f87aef2c6ff1)
